### DISCUSSION TITLE
Is it possible to configure function limits for Next.js API?

### DISCUSSION THREAD

**Discussion by aarshaw** [vercel/vercel#4557] [Help] [answered]
Hi there, our project uses the Next.js API routes (functions under the `pages/api` directory). 

Is it possible to configure `maxDuration` and `memory` for these functions as illustrated here: https://vercel.com/docs/configuration#project/functions

Just realised our functions are using max ~100-200mb but are being deployed with Memory Size of 3008mb - any way to reduce this?

**Comment by mcsdevv** [comment, marked as answer]
Hey @aarshaw, you can use the [functions configuration](https://vercel.com/docs/configuration#project/functions) that you linked 👍

**Reply by aarshaw** [reply]
For this I get: `Error! The function for api/test can't be handled by any builder` where `test.js` is a function in the Next.js API folder `src/pages/api`

```
 "functions": {
    "api/test": {
      "memory": 1024,
      "maxDuration": 30
    }
  }
```

**Reply by mcsdevv** [reply]
Hello @aarshaw, can you try this path? `pages/api/test.js`

**Reply by aarshaw** [reply]
Spot on that's worked 👍
